src/clr-core/internal/base/focus-trap.base.ts

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) 2016-2020 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import { html, LitElement } from 'lit-element';
+import { property } from '../decorators/property';
+import { FocusTrapTracker } from '../services/focus-trap-tracker.service';
+import { isHTMLElement } from '../utils/dom';
+import { isBrowser } from '../utils/exists';
+
+export class CdsBaseFocusTrap extends LitElement {
+  private previousFocus: Element;
+  private topReboundEl: any;
+  private bottomReboundEl: any;
+  private onFocusInEvent: any;
+
+  @property({ attribute: false })
+  private hostElement = this;
+
+  connectedCallback() {
+    super.connectedCallback();
+
+    if (document.activeElement) {
+      this.previousFocus = document.activeElement;
+    }
+
+    // add class to prevent scroll on body
+    document.body.classList.add('no-scrolling');
```

Comment:
Angular version had this as a separate `ScrollingService` class, but it was only being used by modal. I moved this into this base class instead. Let me know if you think if this line of logic (with the corresponding remove) makes more sense to live in Modal class instead.

Reply:
Should update to use cds-layout attr described below